prime-router/src/main/kotlin/azure/EmailEngineFunction.kt

```diff
@@ -136,16 +137,36 @@ class EmailScheduleEngine {
 
             // get the orgs to fire for
             orgs.forEach { org ->
+                val countOfRecords = workflowEngine.db.fetchDownloadableReportFiles(
+                    OffsetDateTime.now().minusDays(1L),
+                    org
+                ).size;     
+                val template = getTemplate(schedule.template, countOfRecords );     
                 val emails: Iterable<String> = if (schedule.emails.size > 0) schedule.emails else getEmails(org, logger)
-                logger.info("EmailEngineFunction:: processing organization $org within ${schedule.template}")
+                logger.info("EmailEngineFunction:: processing organization $org within ${template}")
                 emails.forEach { email ->
-                    logger.info("EmailEngineFunction:: sending email to $email")
-                    dispatchToSendGrid(schedule.template, listOf(email), logger)
+                    logger.info("EmailEngineFunction:: sending email template ${template} to $email")
```

Comment:
Would it make more sense to capture whether the dispatch to sendgrid is successful and then in a `finally` block say: "email to X send, here's the result code of the dispatch")? This may make it easier to debug in the future (which is why we log in the first place)